carina-webdriver/src/main/java/com/qaprosoft/carina/core/foundation/webdriver/core/factory/impl/IOSFactory.java

```diff
@@ -0,0 +1,92 @@
+package com.qaprosoft.carina.core.foundation.webdriver.core.factory.impl;
+
+import java.lang.invoke.MethodHandles;
+import java.net.MalformedURLException;
+import java.net.URL;
+
+import org.openqa.selenium.Capabilities;
+import org.openqa.selenium.Platform;
+import org.openqa.selenium.WebDriver;
+import org.openqa.selenium.remote.Browser;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.qaprosoft.carina.core.foundation.utils.Configuration;
+import com.qaprosoft.carina.core.foundation.webdriver.core.capability.CapabilitiesBuilder;
+import com.qaprosoft.carina.core.foundation.webdriver.core.capability.OptionsType;
+import com.qaprosoft.carina.core.foundation.webdriver.core.factory.AbstractFactory;
+import com.qaprosoft.carina.core.foundation.webdriver.listener.EventFiringAppiumCommandExecutor;
+
+import io.appium.java_client.ios.IOSDriver;
+import io.appium.java_client.remote.AutomationName;
+import io.appium.java_client.remote.options.SupportsAutomationNameOption;
+
+public class IOSFactory extends AbstractFactory {
+    private static final Logger LOGGER = LoggerFactory.getLogger(MethodHandles.lookup().lookupClass());
+
+    @Override
+    public WebDriver create(String testName, String seleniumHost) {
+        return this.create(testName, seleniumHost, null);
+    }
+
+    @Override
+    public WebDriver create(String testName, String seleniumHost, Capabilities capabilities) {
+        CapabilitiesBuilder capabilitiesBuilder = CapabilitiesBuilder.builder();
+        if (capabilities != null) {
+            capabilitiesBuilder.fromCustomCapabilities(capabilities);
+        }
+        capabilitiesBuilder.chooseOptionsType(OptionsType.IOS);
+        Capabilities xcuiOptions = capabilitiesBuilder.build();
+
+        LOGGER.debug("capabilities: {}", xcuiOptions);
+
+        URL hostURL;
+        try {
+            hostURL = new URL(seleniumHost);
+        } catch (MalformedURLException e) {
+            throw new RuntimeException("Malformed selenium URL!", e);
+        }
+
+        EventFiringAppiumCommandExecutor ce = new EventFiringAppiumCommandExecutor(hostURL);
+        IOSDriver driver = new IOSDriver(ce, xcuiOptions);
+
+        registerDevice(driver);
+        return driver;
+    }
+
+    /**
+     * Determines if the driver is suitable for the current capabilities
+     */
+    public static boolean isSuitable(Capabilities capabilities) {
```

Comment:
why do we need this method?

Reply:
this method allows us to determine based on the capability (and in some cases the selenium link type) whether to use this driver (defines the selection rule)